src/network/src/esp/include/NetworkBroadcast.h

```diff
@@ -1,28 +1,36 @@
 #ifndef HIVE_CONNECT_NETWORKBROADCAST_H
 #define HIVE_CONNECT_NETWORKBROADCAST_H
 
+#include "BaseTask.h"
 #include "INetworkBroadcast.h"
+#include "c-common/circular_buff.h"
+#include "logger/ILogger.h"
+#include <array>
 
 class NetworkBroadcast : public INetworkBroadcast {
   public:
-    NetworkBroadcast() = default;
-    ~NetworkBroadcast() override = default;
+    NetworkBroadcast(ILogger& logger);
+    ~NetworkBroadcast() override;
 
-    bool send(const uint8_t* data, uint16_t length) override {
-        (void)data;
-        (void)length;
-        return false;
-    }
+    bool send(const uint8_t* data, uint16_t length) override;
 
-    bool receive(uint8_t* data, uint16_t length) override {
-        (void)data;
-        (void)length;
-        return false;
-    }
+    bool receive(uint8_t* data, uint16_t length) override;
 
-    bool start() override { return false; }
+    bool start() override;
 
-    bool stop() override { return false; }
+    bool stop() override;
+
+    void receiveDatagrams();
+
+  private:
+    ILogger& m_logger;
+    BaseTask<2 * configMINIMAL_STACK_SIZE> m_receivingTask;
+    std::array<uint8_t, 512> m_data;
+    CircularBuff m_circularBuffer;
+    TaskHandle_t m_receivingTaskHandle;
+    int m_socket;
+    bool m_started;
+    std::array<uint8_t, 255> m_datagram;
```

Comment:
Could we parameterize the size of those arrays with a compilation setting? or maybe a define for now

Reply:
I'll add them in the network config header